test-suite-server-generator/src/main/java/io/micronaut/openapi/test/dated/DatedResponseBodyWriter.java

```diff
@@ -66,27 +40,23 @@ public void writeTo(
         MutableHeaders headers,
         OutputStream outputStream
     ) throws CodecException {
-        if (bodyType != null && bodyWriter != null) {
-            if (dated.getLastModified() != null) {
-                headers.add(LAST_MODIFIED_HEADER, dated.getLastModified().toString());
-            }
-            bodyWriter.writeTo(bodyType, mediaType, dated.getBody(), headers, outputStream);
-        } else {
-            throw new ConfigurationException("No JSON message writer present");
+        Argument<T> bt = type.getTypeParameters()[0];
+        MessageBodyWriter<T> writer = registry.findWriter(bt, List.of(MediaType.APPLICATION_JSON_TYPE))
```

Comment:
findWriter should be cached using createSpecific instead of doing it inside writeTo.